lib/addons/ack/index.ts

```diff
@@ -1,122 +1,135 @@
-
-  readonly options: AckAddOnProps;
-  readonly id? : string;
-
-  constructor(props?: AckAddOnProps) {
-    super(populateDefaults(defaultProps, props) as HelmAddOnProps);
-    this.options = this.props as AckAddOnProps;
-    this.id = this.options.id;
-  }
-
-  
-  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
-    const cluster = clusterInfo.cluster;
-
-    const sa = cluster.addServiceAccount(`${this.options.chart}-sa`, {
-      namespace: this.options.namespace,
-      name: this.options.saName,
-    });
-
-    let values: Values = populateValues(this.options,cluster.stack.region);
-    values = merge(values, this.props.values ?? {});
-
-    if(this.options.createNamespace == true){
-      // Let CDK Create the Namespace
-      const namespace = createNamespace(this.options.namespace! , cluster);
-      sa.node.addDependency(namespace);
-    }
-    
-    sa.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName(this.options.managedPolicyName!));
-    const chart = this.addHelmChart(clusterInfo, values);
-    chart.node.addDependency(sa);
-    return Promise.resolve(chart);
-  }
-}
-
-/**
- * populateValues populates the appropriate values used to customize the Helm chart
- * @param helmOptions User provided values to customize the chart
- */
-function populateValues(helmOptions: AckAddOnProps, awsRegion: string): Values {
-  const values = helmOptions.values ?? {};
-  setPath(values, "aws.region", awsRegion);
-  setPath(values,"serviceAccount.create", false);
-  setPath(values,"serviceAccount.name", helmOptions.saName);
-  return values;
-}
-
-/**
- * populate parameters passed or the default values from service Mappings.
- */
-function populateDefaults(defaultProps: AckAddOnProps, props?: AckAddOnProps): AckAddOnProps {
-  let tempProps : Partial<AckAddOnProps> = {...props ?? {}}; // since props may be empty
-  tempProps.id = tempProps.id ?? defaultProps.id;
-  tempProps.serviceName = tempProps.serviceName ?? defaultProps.serviceName;
-  tempProps.name = tempProps.name ?? serviceMappings[tempProps.serviceName!]!.chart;
-  tempProps.namespace = tempProps.namespace ?? defaultProps.namespace;
-  tempProps.chart = tempProps.chart ?? serviceMappings[tempProps.serviceName!]?.chart;
-  tempProps.version = tempProps.version ?? serviceMappings[tempProps.serviceName!]?.version;
-  const repositoryUrl = "oci://public.ecr.aws/aws-controllers-k8s";
-  tempProps.release = tempProps.release ?? tempProps.chart;
-  tempProps.repository = tempProps.repository ?? `${repositoryUrl}/${tempProps.name}`;
-  tempProps.managedPolicyName = tempProps.managedPolicyName ?? serviceMappings[tempProps.serviceName!]?.managedPolicyName;
-  tempProps.createNamespace = tempProps.createNamespace ?? defaultProps.createNamespace;
-  tempProps.saName = tempProps.saName ?? `${tempProps.chart}-sa`;
-  return tempProps as AckAddOnProps;
-}
\ No newline at end of file
+import { ManagedPolicy, Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
```

Comment:
Was there a chance in this file? It is very hard to see because of the formatting. Short of reapplying the changes without formatting, may you can point specific lines that were changed? 
Is it only about allowing to pass inline policies along with managed?

Reply:
@elamaran11 ^^ we will need functional validation before we merge. I can run e2e to make sure no regression, but new capability of passing inline policy will need a separate QA exercise.

Reply:
<img width="582" alt="image" src="https://github.com/aws-quickstart/cdk-eks-blueprints/assets/17073333/d871c3a3-2d19-479e-8212-7d91e88c8f64">
try viewing without whitespace. the issue is the file was encoded with CLRD line endings (windows) which differs from other files in this repo
yes it is only about allowing the passing of inline policies